planetscale/imports.go

```diff
@@ -0,0 +1,321 @@
+	Database     string `json:"name"`
+}
+
+type MakePlanetScaleReplicaRequest struct {
+	Organization string `json:"organization"`
+	Database     string `json:"name"`
+}
+
+type DetachExternalDatabaseRequest struct {
+	Organization string `json:"organization"`
+	Database     string `json:"name"`
+}
+
+type GetImportStatusRequest struct {
+	Organization string `json:"organization"`
+	Database     string `json:"name"`
+}
+
+type CancelDataImportRequest struct {
+	Organization string `json:"organization"`
+	Database     string `json:"name"`
+}
+
+// DataImportsService is an interface for communicating with the PlanetScale
+// Data Imports API endpoint.
+type DataImportsService interface {
+	TestDataImportSource(ctx context.Context, request *TestDataImportSourceRequest) (*TestDataImportSourceResponse, error)
+	StartDataImport(ctx context.Context, request *StartDataImportRequest) (*DataImport, error)
+	CancelDataImport(ctx context.Context, request *CancelDataImportRequest) error
+	GetDataImportStatus(ctx context.Context, request *GetImportStatusRequest) (*DataImport, error)
+	MakePlanetScalePrimary(ctx context.Context, request *MakePlanetScalePrimaryRequest) (*DataImport, error)
+	MakePlanetScaleReplica(ctx context.Context, request *MakePlanetScaleReplicaRequest) (*DataImport, error)
+	DetachExternalDatabase(ctx context.Context, request *DetachExternalDatabaseRequest) (*DataImport, error)
+}
+
+type dataImportsService struct {
+	client *Client
+}
+
+// TestDataImportSource will check an external database for compatibility with PlanetScale
+func (d *dataImportsService) TestDataImportSource(ctx context.Context, request *TestDataImportSourceRequest) (*TestDataImportSourceResponse, error) {
+	request.Source.SSLMode = request.Source.SSLVerificationMode.String()
+	path := fmt.Sprintf("/v1/organizations/%s/data-imports/test-connection", request.Organization)
+	req, err := d.client.newRequest(http.MethodPost, path, request)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	resp := &TestDataImportSourceResponse{}
+	if err := d.client.do(ctx, req, &resp); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (d *dataImportsService) StartDataImport(ctx context.Context, request *StartDataImportRequest) (*DataImport, error) {
+	request.Source.SSLMode = request.Source.SSLVerificationMode.String()
+	path := fmt.Sprintf("/v1/organizations/%s/data-imports/new", request.Organization)
+	req, err := d.client.newRequest(http.MethodPost, path, request)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	resp := &DataImport{}
+	if err := d.client.do(ctx, req, &resp); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (d *dataImportsService) GetDataImportStatus(ctx context.Context, getReq *GetImportStatusRequest) (*DataImport, error) {
+	path := fmt.Sprintf("/v1/organizations/%s/databases/%s", getReq.Organization, getReq.Database)
```

Comment:
Observation: Looking at this, I think it'd make sense to make a separate `/data-import` API endpoint that we can use for just fetching the data import information on its own, I think. This is also beneficial because if the data import doesn't exist, we can then return a 404 from that endpoint.